src/lib/core/diag.h

```diff
@@ -200,6 +200,18 @@ error_set_uuid(struct error *e, const char *name, const struct tt_uuid *value)
 	error_payload_set_uuid(&e->payload, name, value);
 }
 
+static inline const char *
+error_get_mp(const struct error *e, const char *name, uint32_t *size)
```

Comment:
Please add unit tests for this. Better do it in a separate commit.

Reply:
Moved to the separate commit. The added tests a in the diff.

<details>
<summary>diff</summary>

```diff
diff --git a/test/unit/mp_error.cc b/test/unit/mp_error.cc
index f1fd5f4d7..6dfbdce88 100644
--- a/test/unit/mp_error.cc
+++ b/test/unit/mp_error.cc
@@ -461,7 +461,7 @@ static void
 test_payload(void)
 {
 	header();
-	plan(11);
+	plan(13);
 	char buffer[2048];
 	memset(buffer, 0, sizeof(buffer));
 
@@ -479,6 +479,9 @@ test_payload(void)
 	struct tt_uuid uuid;
 	tt_uuid_create(&uuid);
 	error_set_uuid(e, "key6", &uuid);
+	char mp[128];
+	size_t mp_size = mp_format(mp, lengthof(mp), "[%d, %s]", 7, "bye");
+	error_set_mp(e, "key7", mp, mp_size);
 
 	mp_encode_error(buffer, e);
 	error_unref(e);
@@ -507,6 +510,10 @@ test_payload(void)
 	struct tt_uuid val_uuid;
 	ok(error_get_uuid(e, "key6", &val_uuid) &&
 	   tt_uuid_is_equal(&uuid, &val_uuid), "key uuid");
+	uint32_t val_mp_size;
+	const char *val_mp =  error_get_mp(e, "key7", &val_mp_size);
+	ok(val_mp_size == mp_size);
+	ok(val_mp != NULL && memcmp(val_mp, mp, mp_size) == 0);
 
 	error_unref(e);
 
diff --git a/test/unit/mp_error.result b/test/unit/mp_error.result
index 781ae640e..0e9a6b707 100644
--- a/test/unit/mp_error.result
+++ b/test/unit/mp_error.result
@@ -46,7 +46,7 @@ ok 4 - subtests
 ok 5 - subtests
 	*** test_unknown_additional_fields: done ***
 	*** test_payload ***
-    1..11
+    1..13
     ok 1 - code
     ok 2 - errno
     ok 3 - msg
@@ -58,6 +58,8 @@ ok 5 - subtests
     ok 9 - key double
     ok 10 - key bool
     ok 11 - key uuid
+    ok 12 - line 515
+    ok 13 - line 516
 ok 6 - subtests
 	*** test_payload: done ***
 	*** test_mp_print ***

```

</details>